tests/cli/main.cpp

```diff
@@ -0,0 +1,270 @@
+      account_object nathan_acct_after_upgrade = wapiptr->get_account("nathan");
+
+      // verify that the upgrade was successful
+      BOOST_CHECK_PREDICATE( std::not_equal_to<uint32_t>(), (nathan_acct_before_upgrade.membership_expiration_date.sec_since_epoch())(nathan_acct_after_upgrade.membership_expiration_date.sec_since_epoch()) );
+      BOOST_CHECK(nathan_acct_after_upgrade.is_lifetime_member());
+
+      // create a new account
+      graphene::wallet::brain_key_info bki = wapiptr->suggest_brain_key();
+      BOOST_CHECK(!bki.brain_priv_key.empty());
+      signed_transaction create_acct_tx = wapiptr->create_account_with_brain_key(bki.brain_priv_key, "jmjatlanta", "nathan", "nathan", true);
+      // save the private key for this new account in the wallet file
+   	  BOOST_CHECK(wapiptr->import_key("jmjatlanta", bki.wif_priv_key));
+      wapiptr->save_wallet_file(wallet_filename.str());
+
+      // attempt to give jmjatlanta some bitsahres
+      BOOST_TEST_MESSAGE("Transferring bitshares from Nathan to jmjatlanta");
+      signed_transaction transfer_tx = wapiptr->transfer("nathan", "jmjatlanta", "10000", "BTS", "Here are some BTS for your new account", true);
+
+      // get the details for init1
+      witness_object init1_obj = wapiptr->get_witness("init1");
+      int init1_start_votes = init1_obj.total_votes;
+      // Vote for a witness
+      signed_transaction vote_witness1_tx = wapiptr->vote_for_witness("jmjatlanta", "init1", true, true);
+
+      // wait for a maintenance interval
+      // NOTE: For this to work consistently, your maintenance interval must be less than 20 seconds
+      fc::usleep(fc::seconds(20));
+      // send a block to trigger maintenance interval
+      BOOST_CHECK(generate_block(*app1.get()));
+
+      // Verify that the vote is there
+      init1_obj = wapiptr->get_witness("init1");
+      witness_object init2_obj = wapiptr->get_witness("init2");
+      int init1_middle_votes = init1_obj.total_votes;
+      BOOST_CHECK(init1_middle_votes > init1_start_votes);
+
+      // Vote for a 2nd witness
+      int init2_start_votes = init2_obj.total_votes;
+      signed_transaction vote_witness2_tx = wapiptr->vote_for_witness("jmjatlanta", "init2", true, true);
+
+      // send another block to trigger maintenance interval
+      fc::usleep(fc::seconds(20));
+      BOOST_CHECK(generate_block(*app1.get()));
+
+      // Verify that both the first vote and the 2nd are there
+      init2_obj = wapiptr->get_witness("init2");
+      init1_obj = wapiptr->get_witness("init1");
+
+      int init2_middle_votes = init2_obj.total_votes;
+      BOOST_CHECK(init2_middle_votes > init2_start_votes);
+      int init1_last_votes = init1_obj.total_votes;
+      BOOST_CHECK(init1_last_votes > init1_start_votes);
+
+      // set the voting proxy to nathan
+      BOOST_TEST_MESSAGE("About to set voting proxy.");
+      signed_transaction voting_tx = wapiptr->set_voting_proxy("jmjatlanta", "nathan", true);
```

Comment:
make test to check if the proxy changed.

Reply:
actually, i think they should be separated test than the `cli_vote_for_2_witnesses `, something like `proxy_update` will do it.
in `cli_vote_for_2_witnesses` only add the needed stuff for that test. we will add more as we need them in different test cases.

Reply:
Created a new test for testing voting proxy